### PR TITLE
Add an user manual page

### PR DESCRIPTION
**Pull request by yannham**
Add user manual pages. We may want to use something like mdbook in the future, but I went for a pure Gatsby solution for now.

## Content

- Install required packages to get markdown support for gatsby
- Add a sidebar component to hold the table of content
- Get the manual from the `nickel-flake` and display them as `user-manual/xxx`, with code highlighted using PrismJS.
- Update the getting started pages to fix the syntax and improve on the "get nickel running" sections

## Todo in this PR

 - [x] install required packages for markdown support in gatsby
 - [x] add a page for each markdown file as `/user-manual/xxx`
 - [x] enable prisms.js support with the custom Nickel syntax highlighter
 - [x] add a sidebar menu
 - [x] automate the download of the manual from the nickel flake into `nickel-manual`
 - [x] add the required metadata to the manual sections in the `nickel` repo (a `slug` field)
 - [x] add missing introduction and quickstart sections
 - [x] add redirections from "foo.md" to "foo" to make manual's intralinks work
 - [x] make the menu on the left sticky (maybe later)
 - [x] have sub-menus corresponding to subsections (maybe later)
 
## Future TODO

- Update the Nickel syntax definition for Prism (a lot has changed since it has been written)

### CODE REVIEW

File: src/styles/custom.scss [hunk 1]
```diff
@@ -105,4 +105,8 @@ a:hover { text-decoration: none};
   0%   { transform:translate(0,-20px); opacity: 0;  }
   50%  { opacity: 1;  }
   100% { transform:translate(0,20px); opacity: 0; }
+}
+
+.all-links-primary a {
+  @include link-theme("primary");
 }
```
Comment (ErinvanderVeen) [inline src/styles/custom.scss:112]: Missing newline

**Review by ErinvanderVeen — CHANGES_REQUESTED** [review]
- There is a missing ) in "Current state and roadmap", notably about the custom merge functions
- The wrong link to "getting started" at https://622a0f4a313c8144d3bdc420--nickel-lang.netlify.app/user-manual/introduction/#getting-started could be replaced with a relative link to avoid the netlify issue.
- The table headers (for example the Operators, Associativity, Remark table in the syntax chapter) are very close to one another.
- The syntax highlighting is sometimes broken, for instance only the "in" in "builtin" and "min" is highlighted. Could we enforce that "in" starts and ends with a word bound: "<in>"?
- "The reader can find more information about merging in the relevant documentation." It would be nice if there was a link to that documentation somewhere in that sentence.
- Do we want the specifications as part of the user manual? For me, it disrupts the flow of reading.
- There are many places where we have a monospace text, for example in examples, but sometimes also error messages. I think these texts should be more easily distinguishable from the surrounding (non monospace) text. Either by displaying them with a dark background, like we do for some of the code examples, or at least surrounded by a border.
- There is a (leftover footmark) in the correctness chapter: "This statement is to be nuanced. Using the Nickel LSP server in your"
- The "User-defined contracts" should have some text below it. Perhaps: "There are two ways to construct contracts: by hand, and by using the `from_predicate` function"

**Comment by yannham** [comment]
Thanks for reviewing! I'll look into it.

As a preliminary answers:

- The text of the manual is directly taken from the Nickel repo, so fixing text is (surprisingly, maybe) out of scope of this PR. But that's till useful information to patch the nickel-repo directly.
- Syntax highlighting has to be updated, yeah, but I think I'll do that in a subsequent PR. It's not specific to the manual.
- I agree about table headers, but I use a generic Gatsby plugin to transform md files to HTML. I'll take a quick look if it is easily customizable, but I don't plan to spend too much time on tweaking the default.
- The wrong link to `Getting Started` can't be fixed: it comes from the manual inside the Nickel repo, which is distributed as standalone MD files that could be read from anywhere. Hence, we need an absolute link to make it work inside MD files. That's also fine IMHO because once this PR is merged, it becomes an non-issue.

File: src/pages/getting-started.js [hunk 2]
```diff
@@ -100,11 +115,29 @@ const IndexPage = () => {
...
+                <p>With a recent version of Nix (> 2.4.0), you can build an run Nickel in one shot. If you haven't installed Nix yet, please follow
+                    <a className={"link-primary"} href={"https://nixos.org/guides/nix-pills/install-on-your-running-system.html"}> this installation guide</a>. Once Nix is installed, use <code>nix run</code> to start Nickel
+                    and append <code>-- args</code> to pass arguments to the Nickel executable (here we launch an REPL session)</p>
+
+                <pre className={'command-line language-bash'} data-user="devops" data-host="nickel"
+                     data-output="2:"><code>{codeExamples.withNix.nix_run}</code></pre>
+
+                <h3>Build from source</h3>
+
+                <p>If you are planning to try Nickel a bit more extensively, you can build the binary to avoid having to go through Nix at each execution (works for Nix versions prior to 2.4.0 too):</p>
```
[the remark this answers is not in the document]
Comment (yannham) [reply src/pages/getting-started.js:140]: Good suggestion

File: src/pages/user-manual/{MarkdownRemark.frontmatter__slug}.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        headings
+    };
+  return (
+      <Layout>
+        <div className="container-fluid">
+            <div className={"row"}>
+                {/*col-md-4 col-lg-3 col-xl-3*/}
+                <div className={"col-xl-3 col-lg-4 col-md-5 order-1"}>
+                    <SidebarToc {...sidebarProps}/>
+                </div>
```
[the remark this answers is not in the document]
Comment (yannham) [reply src/pages/user-manual/{MarkdownRemark.frontmatter__slug}.js:32]: That's strange, I don't experience the same, whatever I try.

**Comment by yannham** [comment]
And now rebase on the grammar fix.

**Review by ErinvanderVeen — APPROVED** [review]

**Pull request merged**